tensorbay/utility/common.py

```diff
@@ -85,3 +86,41 @@ def wrapper(self: Any, *arg: Any, **kwargs: Any) -> None:
             lock.release()
 
     return wrapper  # type: ignore[return-value]
+
+
+manager = Manager()
+process_locks: Dict[str, Lock] = manager.dict()
+
+
+def get_process_locked(
+    attr_name: str = "",
+) -> Callable[[_CallableWithoutReturnValue], _CallableWithoutReturnValue]:
+    """Get the decorator to add lock for methods called from different processes.
+
+    Arguments:
+        attr_name: The name of the attr to be taken as the key of the lock.
+
+    Returns:
+        The decorator to add lock for methods called from different processes.
+
+    """
+
+    def process_locked(func: _CallableWithoutReturnValue) -> _CallableWithoutReturnValue:
+        @wraps(func)
+        def wrapper(self: Any, *arg: Any, **kwargs: Any) -> None:
+            key = getattr(self, attr_name) if attr_name else id(self)
+            # https://github.com/PyCQA/pylint/issues/3313
+            lock = process_locks.setdefault(key, manager.Lock())  # pylint: disable=no-member
```

Comment:
get item from a Manager.dict() seems a copy behavior, which means the gotten lock is a copy of the lock in the dict. That means the operation in the outside lock will not affect the inside lock?

Reply:
Shared objects are capable of being nested. The `manager.Lock()` is also a shared object, thus the change will be synced to the dict.
https://docs.python.org/3.8/library/multiprocessing.html#multiprocessing.managers.SyncManager